infra/scripts/local-ci.sh

```diff
@@ -0,0 +1,198 @@
+  echo "  variables.  If a secrets file is not provided, this script will attempt to retrieve some info from govc, such as"
+  echo "  TEST_URL, TEST_USERNAME, and TEST_PASSWORD."
+  echo
+  echo "  options:"
+  echo "    -t DOCKER_TEST (or env var)"
+  echo "    -v TARGET_VCH (or env var)"
+  echo "    -s SECRETS_FILE (or env var)"
+  echo "    -g GITHUB_API_KEY (or env var)"
+  echo "    -u TEST_URL (or env var or in secretsfile)"
+  echo "    -d TEST_DATASTORE (or env var or in secretsfile)"
+  echo "    -n TEST_USERNAME (or env var or in secretsfile)"
+  echo "    -p TEST_PASSWORD (or env var or in secretsfile)"
+  echo
+  echo "  example:"
+  echo "    $0 -t Group1-Docker-Commands -v my_vch -s test.secrets.nested -g xxxxxx"
+  echo "    $0 -t 1-01-Docker-Info.robot -v my_vch -s test.secrets.nested -g xxxxxx"
+  echo 
+  echo "    DOCKER_TEST=Group1-Docker-Commands/1-01-Docker-Info.robot TARGET_VCH=my_vch SECRETS_FILE=test.secrets.nested $0"
+  echo
+  echo "    $0 -s test.secrets.nested (all params defined in secrets file)"
+  exit 1
+}
+
+function GetGovcParamsFromEnv() {
+  echo "Getting params from GOVC var"
+  test_username=$(govc env | grep GOVC_USERNAME | cut -d= -f2)
+  test_password=$(govc env | grep GOVC_PASSWORD | cut -d= -f2)
+  test_url=$(govc env | grep GOVC_URL | cut -d= -f2)
+}
+
+function GetParamsFromSecrets() {
+  echo "Getting params from the secrets file"
+  secrets_api_key="$(grep 'GITHUB_AUTOMATION_API_KEY' ${secretsfile} | awk '{ print $2 }')"
+  secrets_url="$(grep 'TEST_URL_ARRAY' ${secretsfile} | awk '{ print $2 }')"
+  secrets_datastore="$(grep -E '\s+TEST_DATASTORE' ${secretsfile} | awk '{ print $2 }')"
+  secrets_username="$(grep 'TEST_USERNAME' ${secretsfile} | awk '{ print $2 }')"
+  secrets_password="$(grep 'TEST_PASSWORD' ${secretsfile} | awk '{ print $2 }')"
+  secrets_docker_test="$(grep 'DOCKER_TEST' ${secretsfile} | awk '{ print $2 }')"
+  secrets_target_vch="$(grep 'TARGET_VCH' ${secretsfile} | awk '{ print $2 }')"
+}
+
+##### Get command line arguments
+while getopts s:t:v:g:u:d:n:p: flag
+do
+  case $flag in
+    s)
+      secretsfile=$OPTARG
+      ;;
+    t)
+      docker_test="$OPTARG"
+      ;;
+    v)
+      target_vch=$OPTARG
+      ;;
+    g)
+      github_api_key=$OPTARG
+      ;;
+    u)
+     test_url=$OPTARG
+      ;;
+    d)
+      test_datastore=$OPTARG
+      ;;
+    n)
+      test_username=$OPTARG
+      ;;
+    p)
+      TEST_PASSWORD=$OPTARG
+      ;;
+    *)
+      usage
+      ;;
+  esac
+done
+
+##### Preconditions... 
+if [[ -z "${SECRETS_FILE}" ]] && [[ -z $secretsfile ]] ; then
+  GetGovcParamsFromEnv
+elif [[ -z $secretsfile ]] ; then
+  secretsfile=${SECRETS_FILE}
+  GetParamsFromSecrets
+else
+  GetParamsFromSecrets
+fi
+
+if [[ -z "${TARGET_VCH}" ]] && [[ -z $target_vch ]] && [[ -z $secrets_target_vch ]]; then 
+  usage
+elif [[ -z $target_vch ]] ; then
+  target_vch=${TARGET_VCH}
```

Comment:
I think this needs to be:
```
target_vch=${TARGET_VCH:-$secrets_target_vch}
```
and the whole may be easier to write as:
```
target_vch=${TARGET_VCH:-$secrets_target_vch}
if [[ -z "${target_vch}" ]]; then
   usage
fi
```

same conciseness comment applies to the rest of the tests.